### ISSUE TITLE
Adjust coding conventions on braces

### ISSUE THREAD

**Issue by dariusf** [HubTurbo/HubTurbo#863]
We want to make braces mandatory but allow them to be on the same line.

```java
// Allowed

if (true) { return 1; }

// Not allowed

if (true) {return 1;} // Pad with spaces

if (true)
    return 1;

if (true) return 1;
```

This issue entails:

- [x] Modifying the CheckStyle configuration
- [x] Fixing all violations that result
- [x] Updating the coding conventions document

Related to #745, context in #861.

**Comment by codemanmeet** [comment]
@damithc @dariusf  Single line block allowance is only available from [Checkstyle 6.9 onwards](https://github.com/checkstyle/checkstyle/issues/1019). Also, the current Gradle version (2.6) doesn't support Checkstyle 6.8+ due to an existing [bug](https://discuss.gradle.org/t/checkstyle-plugin-does-not-work-with-cs-6-8/10353) that has been [fixed](https://github.com/gradle/gradle/pull/491) and will be available only in [Gradle 2.7](https://github.com/gradle/gradle/pull/491#issuecomment-129502919).

On a side note, I feel that braces should be compulsory for all multi-line blocks. [Google Java coding conventions](https://source.android.com/source/code-style.html) state that the following is legal - 

`if (condition) body();`

What do you'll think?

I can make braces compulsory for all `if`, `else`, `while` and `do` statements except if they are single line blocks, which can only be done for Checkstyle 6.8+. So shall I go with braces compulsory for all multi-line blocks in this sprint?

**Comment by damithc** [comment]
I'm OK with that. @dariusf?

**Comment by dariusf** [comment]
Yup, @codemanmeet told me about this already. We were just concerned whether this reflected TEAMMATES' standards, since we want to target them eventually.

**Comment by damithc** [comment]
In the worst case, we can tweak teammates standard to match this :-)

**Comment by dariusf** [comment]
Alright then, I'm fine with this as well. If we want to revisit this we can when Gradle 2.7 is released and we can use CheckStyle 6.9.
